File: packages/typia/src/transformers/features/json/JsonCreateValidateParseTransformer.ts
import { JsonValidateParseProgrammer } from "../../../programmers/json/JsonValidateParseProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonCreateValidateParseTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "json.createValidateParse",
      write: JsonValidateParseProgrammer.write,
    });
}

File: packages/typia/native/transform/features/json/JsonCreateValidateParseTransformer.go
package json

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativejsonprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/json"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type jsonCreateValidateParseTransformerNamespace struct{}

var JsonCreateValidateParseTransformer = jsonCreateValidateParseTransformerNamespace{}

func (jsonCreateValidateParseTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "json.createValidateParse",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativejsonprogrammers.JsonValidateParseProgrammer.Write(x)
		},
	})
}
